### PR TITLE
add tests for afex and more

### PR DESCRIPTION
**Pull request by mattansb**
I'll be working here on adding support for `afex_aov`:

- [x] normality
  - [x] also fix in `see` (need to store residuals in object)
  - [x] ~also for multivariate in general~ (already existed)
- [x] homogeneity
- [x] sphericity
  - [x] also for multivariate in general w/ `car`?

### CODE REVIEW

File: R/check_sphericity.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#' @return Invisibly returns the p-values of the test statistics. A p-value < 0.05
+#' indicates a violation of sphericity.
+#'
+#' @export
+check_sphericity <- function(x, ...) {
+  UseMethod("check_sphericity")
+}
+
+#' @export
+check_sphericity.default <- function(x, ...) {
```
Comment (strengejacke) [inline R/check_sphericity.R:18]: I would use `class(X)[1]`, because some model objects inherit from multiple classes.

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by mattansb** [comment]
``` r
library(afex)
library(performance)

data(obk.long, package = "afex")

obk.long$treatment <- as.character(obk.long$treatment)

aM <- aov_car(value ~ treatment * gender + Error(id/(phase*hour)), 
              data = obk.long)

aW <- aov_car(value ~ Error(id/(phase*hour)), 
              data = obk.long)

aB <- aov_car(value ~ treatment * gender + Error(id), 
              data = obk.long)

check_collinearity(aM)
#> All predictors have been centered (factors with contr.sum(), numerics with scale())
#> # Check for Multicollinearity
#> 
#> Low Correlation
#> 
#>                         Term  VIF Increased SE Tolerance
#>                    treatment 1.04         1.02      0.96
#>                       gender 1.07         1.04      0.93
#>                        phase 1.15         1.07      0.87
#>                         hour 1.33         1.15      0.75
#>             treatment:gender 1.07         1.04      0.93
#>              treatment:phase 1.20         1.10      0.83
#>                 gender:phase 1.15         1.07      0.87
#>               treatment:hour 1.45         1.20      0.69
#>                  gender:hour 1.33         1.15      0.75
#>                   phase:hour 1.77         1.33      0.56
#>       treatment:gender:phase 1.20         1.10      0.83
#>        treatment:gender:hour 1.45         1.20      0.69
#>         treatment:phase:hour 2.09         1.45      0.48
#>            gender:phase:hour 1.77         1.33      0.56
#>  treatment:gender:phase:hour 2.09         1.45      0.48

check_collinearity(aW)
#> All predictors have been centered (factors with contr.sum(), numerics with scale())
#> # Check for Multicollinearity
#> 
#> Low Correlation
#> 
#>        Term  VIF Increased SE Tolerance
#>       phase 1.00         1.00      1.00
#>        hour 1.00         1.00      1.00
#>  phase:hour 1.00         1.00      1.00

check_collinearity(aB)
#> All predictors have been centered (factors with contr.sum(), numerics with scale())
#> # Check for Multicollinearity
#> 
#> Low Correlation
#> 
#>              Term  VIF Increased SE Tolerance
#>         treatment 1.04         1.02      0.96
#>            gender 1.07         1.04      0.93
#>  treatment:gender 1.07         1.04      0.93



check_normality(aM)
#> OK: residuals appear as normally distributed (p = 0.205).
check_normality(aW)
#> OK: residuals appear as normally distributed (p = 0.550).
check_normality(aB)
#> OK: residuals appear as normally distributed (p = 0.735).



# check_homogeneity(aW) # fail - no between groups
check_homogeneity(aM)
#> OK: Variances in each of the groups are the same (Levene's Test, p = 0.350).
check_homogeneity(aB)
#> OK: Variances in each of the groups are the same (Levene's Test, p = 0.350).



check_sphericity(aM)
#> Warning: Sphericity violated for: 
#>  - hour (p = 0.008)
#>  - treatment:hour (p = 0.008)
#>  - gender:hour (p = 0.008)
#>  - treatment:gender:hour (p = 0.008).
check_sphericity(aW)
#> Warning: Sphericity violated for: 
#>  - hour (p < .001)
#>  - phase:hour (p = 0.027).
# check_sphericity(aB) # fail - no withing conditions
```

<sup>Created on 2021-04-18 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by mattansb** [comment]
@strengejacke What do you think? Any changes needed?

(I also fixed the qqplot in see to work)

File: R/check_homogeneity.R [hunk 2]
```diff
@@ -55,7 +55,8 @@ check_homogeneity.default <- function(x, method = c("bartlett", "fligner", "leve
   if (method == "auto") {
     check <- tryCatch(
       {
-        stats::shapiro.test(insight::get_residuals(x))$p.value
+        capture.output(p <- check_normality(x))
```
Comment (strengejacke) [inline R/check_homogeneity.R]: please add `utils::`

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/check_collinearity.R [hunk 3]
```diff
@@ -99,6 +99,27 @@ check_collinearity.default <- function(x, verbose = TRUE, ...) {
...
+  # f <- insight::find_formula(x)[[1]]
+  # f <- Reduce(paste, deparse(f))
+  # f <- sub("\\+\\s*Error\\(.*\\)$", "", f)
+  # f <- as.formula(f)
+
+  d <- insight::get_data(x)
+  is_num <- sapply(d, is.numeric)
+  d[is_num] <- sapply(d[is_num], scale, center = TRUE, scale = FALSE)
+  is_fac <- sapply(d, is.factor) | sapply(d, is.character)
+  contrs <- lapply(is_fac, function(...) contr.sum)[is_fac]
```
Comment (strengejacke) [inline R/check_collinearity.R]: need to change to `stats::contr.sum` here.

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by mattansb** [comment]
Anything else?

If not will add some tests and merge $_$

**Comment by strengejacke** [comment]
picard_make_it_so

**Comment by IndrajeetPatil** [comment]
All checks are still failing because they can't find `afex` in `Imports`, so I guess there is a remaining instance where `afex` is not being used conditionally.

**Comment by strengejacke** [comment]
This branch is not in line with master. Please remove all `@importFrom` directives.

File: R/check_collinearity.R [hunk 4]
```diff
@@ -99,6 +99,29 @@ check_collinearity.default <- function(x, verbose = TRUE, ...) {
   .check_collinearity(x, component = "conditional", verbose = verbose)
 }
 
+#' @export
+#' @importFrom stats contr.sum lm
+#' @importFrom insight find_response get_data
```
Comment (strengejacke) [inline R/check_collinearity.R]: please remove all `importFrom`, as was done in master...

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/check_sphericity.R [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  } else {
+    pp <- insight::format_p(min(p.val))
+    pp <- sub("=", ">", pp)
+    insight::print_color(sprintf("OK: Data seems to be spherical (%s).\n", pp), "green")
+  }
+  invisible(p.val)
+}
+
+#' @export
+#' @importFrom insight print_color format_p
```
Comment (strengejacke) [inline R/check_sphericity.R]: Here as well (remove importFrom)

**Review by strengejacke — APPROVED** [review]

**Pull request merged**